modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/mini_cart/MiniCart.js

```diff
@@ -108,61 +132,148 @@ function MiniCart({
 	return (
 		<MiniCartContext.Provider
 			value={{
-				AJAX,
+				CartResource,
+				CartViews,
 				actionURLs,
 				cartState,
 				closeCart,
 				displayDiscountLevels,
+				displayTotalItemsQuantity,
 				isOpen,
 				isUpdating,
+				labels: normalizePartialObject(DEFAULT_LABELS, labels),
 				setIsUpdating,
 				spritemap,
+				summaryDataMapper,
+				toggleable,
 				updateCartModel,
 				updateCartState,
 			}}
 		>
-			{!!CartView && (
-				<div className={classnames('mini-cart', isOpen && 'is-open')}>
-					<div
-						className={'mini-cart-overlay'}
-						onClick={() => setIsOpen(false)}
-					/>
+			{!!CartViews[CART] && (
+				<div
+					className={classnames(
+						'mini-cart',
+						(!toggleable || isOpen) && 'is-open'
```

Comment:
There are lots of different ways to write this, but using the object format might make it more readable for cases like this where the conditional starts to get more complicated:

```js
classnames({
    'mini-cart': true,
    'is-open': isOpen || !toggleable,
})
```

ie. because you can read it from left-to-right as "use class `is-open` if you're open or not toggleable".